### ISSUE TITLE
Malformed URIs passed to `decodeURIComponent` cause integration crash

### ISSUE THREAD

**Issue by sarus** [polarityio/decoder#3]
### Steps to Test

1. Put integration into on-demand mode
2. Run On-Demand on the string `%C5`
3. Integration will crash with the following error:

```
{ stderrString:
   '/home/ed/workspace/polarity-server/integrations/decoder/integration.js:44
let ascii = decodeURIComponent(string);
URIError: URI malformed\n    at decodeURIComponent (<anonymous>)
at _decodeUrlString (/home/ed/workspace/polarity-server/integrations/decoder/integration.js:44:15)
at entities.forEach (/home/ed/workspace/polarity-server/integrations/decoder/integration.js:20:23)
at Array.forEach (<anonymous>)
at Object.doLookup (/home/ed/workspace/polarity-server/integrations/decoder/integration.js:4:12)
at Object.performLookup [as exec] (/home/ed/workspace/polarity-server/lib/utils/integration-loader.js:923:15)
at /home/ed/workspace/polarity-server/lib/utils/integration-loader.js:1114:16
at Array.forEach (<anonymous>)
at process.<anonymous> (/home/ed/workspace/polarity-server/lib/utils/integration-loader.js:1110:14)\n    at process.emit (events.js:182:13)' }
```

**Comment by psiphon** [comment]
`(?:%[0-9][0-9A-Fa-f])|(?:%[A-Fa-f][0-9A-Fa-f]%[0-9A-Fa-f][0-9A-Fa-f])+`
Testing this
https://regex101.com/r/Biy8Ch/1

**Comment by psiphon** [comment]
https://regex101.com/r/Biy8Ch/3
This one requires fewer steps

`%(?:[0-9][0-9A-Fa-f]|[A-Fa-f][0-9A-Fa-f]%[0-9A-Fa-f][0-9A-Fa-f])`

**Comment by sarus** [comment]
Ran a quick test on the last regex and it appears to break up a URL encoded string into it's individual characters where each character is a separate entity when passed to the integration. So for example the following string:
```
%50%4F%4C%41%52%49%54%59%20%52%4F%43%4B%53%21%21%21
```
will become 10 or so separate letters rather than the fully formed string: `Polarity Rocks!`

**Comment by psiphon** [comment]
Wrapped and added a +
https://regex101.com/r/Biy8Ch/7
`(?:%(?:[0-9][0-9A-Fa-f]|[A-Fa-f][0-9A-Fa-f]%[0-9A-Fa-f][0-9A-Fa-f]))+`

**Comment by psiphon** [comment]
This one optimizes a bit and lowers the number of requried steps required to parse
https://regex101.com/r/Biy8Ch/8
`(?:%[0-9][0-9A-Fa-f]|%[A-Fa-f][0-9A-Fa-f]%[0-9A-Fa-f][0-9A-Fa-f])+`

**Comment by sarus** [comment]
So far the last regex is working really well.

**Comment by psiphon** [comment]
I have it on the testing server now. I'll create a PR for it.

**Comment by sarus** [comment]
Fixed in 3.0.2-beta

https://github.com/polarityio/decoder/releases/tag/3.0.2-beta
